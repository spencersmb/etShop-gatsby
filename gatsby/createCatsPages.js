const path = require("path")

module.exports = async ({ actions, graphql }) => {
  const { createPage } = actions
  const perPage = 10
  const allCats = {}
  const excludeItems = ["uncategorized"]
  let pageNumber = 1

  const fetchCatPages = async (variables) =>
    await graphql(GET_CATS, variables).then(async ({ data }) => {
      const {
        wpgraphql: {
          categories: {
            nodes
          }
        }
      } = data

      await Promise.all(nodes
        .filter(node => !excludeItems.includes(node.slug))
        .map((cat) => {
          const {
            supportQuestions: {
              pageInfo: { hasNextPage, endCursor, hasPreviousPage }
            }
          } = cat
          const questions = cat.supportQuestions.nodes
          const catPageTemplate = path.resolve(`./src/templates/supportCatListPage.tsx`)
          const catPagePath = `/support/category/${cat.slug}/page/${pageNumber}`
          // console.log("cat.supportQuestions.pageInfo", cat.supportQuestions.pageInfo)
          // console.log("slug", cat.slug)
          // console.log("ID", cat.id)
          // console.log("pageNumber", pageNumber)
          // console.log("hasNextPage", hasNextPage)
          // console.log("hasPrevPage", hasPreviousPage)

          // add item to be excluded if no next page
          if (!hasNextPage) {
            excludeItems.push(cat.slug)
          }
          const prevPages = allCats[cat.slug] && allCats[cat.slug].pages ? allCats[cat.slug].pages : {}
          // check if object exists first
          allCats[cat.slug] = {
            pages: {
              ...prevPages,
              [pageNumber]: {
                path: catPagePath,
                component: catPageTemplate,
                context: {
                  cat: cat.name,
                  catId: cat.id,
                  questions,
                  pageNumber,
                  hasNextPage,
                  hasPrevPage: hasPreviousPage
                }
              }
            }
          }

          if (hasNextPage) {
            pageNumber++
            return fetchCatPages({ first: perPage, after: endCursor })
          }
        }))
      return allCats
    })

  await fetchCatPages({ first: perPage, after: null }).then(catItems => {
    Object.keys(catItems).map((cat) => {
      const pages = Object.keys(catItems[cat].pages)
      pages.map(page => {
        console.log(`create Cat Page ${cat}`)
        console.log(`create Cat PageNumber ${catItems[cat].pages[page].context.pageNumber}`)
        createPage(catItems[cat].pages[page])
      })
    })
  })
}

const GET_CATS = `
  query GET_ALL_CATS($first:Int $after:String){
      wpgraphql{
        categories {
          nodes {
            id
            name
            slug
            supportQuestions(after: $after, first: $first) {
              pageInfo{
                  hasNextPage
                  hasPreviousPage
                  endCursor
              }
              nodes {
                  title
                  id
                  slug
                  excerpt
                  acfSupportQuestions{
                      popularity
                  }
              }
            }
          }
        }
      }
    }
  `
