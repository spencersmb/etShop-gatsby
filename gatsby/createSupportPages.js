const path = require(`path`)
module.exports = async ({ actions, graphql }) => {
  const supportQuery = `
  {
  wpgraphql {
    supportQuestions(where: {status: PUBLISH}, first: 100){
      nodes{
        date
        uri
        id
        slug
        title
        content
        featuredImage {
          altText
          caption
          sourceUrl
        }
      }
    }
  }
}
  `

  return new Promise((resolve, reject) => {
    graphql(supportQuery).then(results => {

      if (results.errors) {
        console.log(results.errors)
        reject(results.errors)
      }
      results.data.wpgraphql.supportQuestions.nodes.forEach(question => {
        actions.createPage({
          path: `/support/${question.slug}/`,
          component: path.resolve(`./src/templates/supportQuestion.tsx`),
          context: {
            // blogId: blog.id,
            ...question
          }
        })
        console.log(`Created page: /support/${question.slug}`)
      })
      resolve()
    })

  })
}
