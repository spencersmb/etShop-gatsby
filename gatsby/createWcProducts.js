const path = require("path")
const GET_PAGES = `
  query GET_WC_PAGES{
      allWcProduct{
        edges{
          node{
            slug
            name
          }
        }
      }
    }
  `
module.exports = async ({ actions, graphql }) => {
  const { createPage } = actions
  const allPages = []
  const fetchPages = async () =>
    await graphql(GET_PAGES).then(({ data }) => {
      const {
        allWcProduct: {
          edges
        }
      } = data
      edges.map(page => {
        allPages.push(page.node)
      })
      return allPages
    })

  await fetchPages().then(allPages => {
    allPages.map(page => {
      console.log(`create page: ${page.slug}`)
      createPage({
        path: `/products/${page.slug}`,
        component: path.resolve(`./src/templates/productPage.tsx`),
        context: {
          slug: page.slug
        }
      })
    })
  })

  // return new Promise((resolve, reject) => {
  //   graphql(pageQuery).then(results => {
  //
  //     if (results.errors) {
  //       console.log(results.errors)
  //       reject(results.errors)
  //     }
  //     results.data.allWcProduct.edges.forEach(({ node }) => {
  //       createPage({
  //         path: `/products/${node.slug}`,
  //         component: path.resolve(`./src/components/products/productDetailPage.tsx`),
  //         context: {
  //           slug: node.slug
  //         }
  //       })
  //       console.log(`Created page: /products/${node.slug}`)
  //     })
  //   })
  //   resolve()
  // })
}
