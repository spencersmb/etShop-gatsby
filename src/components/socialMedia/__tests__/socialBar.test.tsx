import FeatureItem from '@components/products/modules/features/featureItem'
import SocialBarItem from '@components/socialMedia/socialBarItem'
import { StaticQuery } from 'gatsby'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	type: 'youtube-tuts',
	cat: 'youtube',
	title: 'Looking for free tutorials?',
	description: 'You can also find me every Tuesday sharing a new design, Procreate or lettering tutorial on YouTube',
	svg: 'Youtube',
	buttonText: 'watch',
	link: 'https://youtube.com/everytues',
	svgColor: '#fff'
}
const propsImage = {
	type: 'procreateBeg-course',
	cat: 'Free Course',
	title: 'Procreate for Beginners',
	description: 'If youâ€™re new to Procreate, check out my free course, Procreate for Beginners!',
	svg: 'Youtube',
	buttonText: 'Enroll',
	link: 'https://every-tuesday.com/procreate-for-beginners',
	svgColor: '#fff',
	staticImagePath: 'procreateLogo.png'
}
describe('Feature Item Test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<SocialBarItem {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct title', () => {
		const modalRender = render(<SocialBarItem {...propsDefault}/>)
		expect(modalRender.getByTestId('title').innerHTML).toEqual(propsDefault.title)
	})
	it('Should render correct description', () => {
		const modalRender = render(<SocialBarItem {...propsDefault}/>)
		expect(modalRender.getByTestId('desc').innerHTML).toEqual(propsDefault.description)
	})
	it('Should render correct cat', () => {
		const modalRender = render(<SocialBarItem {...propsDefault}/>)
		expect(modalRender.getByTestId('cat').innerHTML).toEqual(propsDefault.cat)
	})
	it('Should render correct icon', () => {
		const modalRender = render(<SocialBarItem {...propsDefault}/>)
		expect(modalRender.queryAllByTestId('icon').length).toEqual(1)
	})
	it('Should render correct btn', () => {
		const modalRender = render(<SocialBarItem {...propsDefault}/>)
		const btn = modalRender.getByTestId('btn')
		expect(btn.innerHTML).toEqual(propsDefault.buttonText)
		expect(btn.getAttribute('href')).toEqual(propsDefault.link)
	})
	it('Should render an Image', () => {
		// @ts-ignore
		StaticQuery.mockImplementationOnce((mock: any) =>
			mock.render({
				allFile: {
					edges: [
						{
							node: {
								name: 'Dreamy Ink Textures',
								id: '46e80e85-da3a-5a1e-9ef3-bd46e38d1dd5',
								relativePath: 'procreateLogo.png',
								childImageSharp: {
									fluid: {
										aspectRatio: 12,
										srcSet: 'srcSet',
										sizes: '',
										src: '/static/dreamy-textures-800x500.jpg'
									}
								}
							}
						}
					]
				}
			})
		)
		const modalRender = render(<SocialBarItem {...propsImage}/>)
		expect(modalRender.queryAllByTestId('img').length).toEqual(1)
	})
})
