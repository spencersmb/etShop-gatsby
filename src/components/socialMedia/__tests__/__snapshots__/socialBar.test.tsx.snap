// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Item Test renders correctly 1`] = `
<div
  className="sc-bdVaJa jyvzgi"
  data-testid="item"
>
  <div
    className="sc-bwzfXH sc-htpNat fHXyX"
    color="#fff"
    data-testid="icon"
    type="youtube-tuts"
  >
    <test-file-stub />
  </div>
  <div
    className="sc-ifAKCX eUWlWz"
  >
    <div
      className="sc-bZQynM bBieGp"
      data-testid="cat"
    >
      youtube
    </div>
    <h4
      className="sc-EHOje eDbdtz"
      data-testid="title"
    >
      Looking for free tutorials?
    </h4>
    <div
      className="sc-gzVnrw bxTmTy"
      data-testid="desc"
    >
      You can also find me every Tuesday sharing a new design, Procreate or lettering tutorial on YouTube
    </div>
  </div>
  <a
    className="sc-htoDjs jlYWml"
    data-testid="btn"
    href="https://youtube.com/everytues"
    rel="noreferrer"
    target="_blank"
  >
    watch
  </a>
</div>
`;
